### PR TITLE
Refactor log4j core test routing

### PR DESCRIPTION
**Pull request by Realtin**
[A clear and concise description of what the pull request is for along with a reference to the associated issue IDs, if they exist.]

- [ ] DefaultRouteScriptAppenderTest.java (Difficult!)
- [x] JsonRoutingAppender2Test.java
- [x] JsonRoutingAppenderTest
- [x] PropertiesRoutingAppenderTest
- [ ] RoutesScriptAppenderTest (Difficult!)
- [x] RoutingAppender2767Test (Easy)
- [x] RoutingAppender3350Test (Easy)
- [x] RoutingAppenderKeyLookupEvaluationTest (Middle)
- [x] RoutingAppenderTest (Easy)
- [ ] RoutingAppenderWithJndiTest (Middle)
- [ ] RoutingAppenderWithPurgingTest (Middle)
- [x] RoutingDefaultAppenderTest (Easy)

## Checklist

* Base your changes on `2.x` branch if you are targeting Log4j 2; use `main` otherwise
* `./mvnw verify` succeeds (if it fails due to code formatting issues reported by Spotless, simply run `./mvnw spotless:apply` and retry)
* Non-trivial changes contain an entry file in the `src/changelog/.2.x.x` directory
* Tests for the changes are provided
* [Commits are signed](https://docs.github.com/en/authentication/managing-commit-signature-verification/signing-commits) (optional, but highly recommended)

### CODE REVIEW

File: log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/routing/PropertiesRoutingAppenderTest.java [hunk 1]
```diff
@@ -16,63 +16,65 @@
...
+        this.context = context;
+    }
 
-    @Before
-    public void setUp() throws Exception {
-        this.app = this.loggerContextRule.getListAppender("List");
+    @BeforeEach
+    public void beforeEach() throws Exception {
+        new CleanFiles(UNKNOWN_LOG_FILE, ALERT_LOG_FILE, ACTIVITY_LOG_FILE);
+        this.app = context.getConfiguration().getAppender("List");
```
Comment (Realtin) [inline log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/routing/PropertiesRoutingAppenderTest.java]: Line 55 was in a setup function. now I'm not sure if it should have stayed in it's own function with a beforeEach or if it is ok to be in there together with the clean files function?

File: log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/routing/PropertiesRoutingAppenderTest.java [hunk 2]
```diff
@@ -16,63 +16,65 @@
...
     }
 
-    @After
-    public void tearDown() throws Exception {
+    @AfterEach
+    public void tearDown(@Named("List") final ListAppender appender) throws Exception {
         this.app.clear();
-        this.loggerContextRule.getLoggerContext().stop();
+        this.app.stop();
```
Comment (Realtin) [inline log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/routing/PropertiesRoutingAppenderTest.java]: I'm not sure about this.

File: log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/routing/PropertiesRoutingAppenderTest.java (hunk 1 above)
Comment (AlbaHerrerias) [reply log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/routing/PropertiesRoutingAppenderTest.java]: This looks correct to me 👍  Maybe add the missing `this` (`this.context` in line 55)

File: log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/routing/PropertiesRoutingAppenderTest.java (hunk 2 above)
Comment (AlbaHerrerias) [reply log4j-core-test/src/test/java/org/apache/logging/log4j/core/appender/routing/PropertiesRoutingAppenderTest.java]: As I understand this is not equivalent. `app` is an appender, and `this.loggerContextRule.getLoggerContext().stop();` is a logger. We can debug this next day

**Comment by Realtin** [comment]
closing in favour of https://github.com/neighbourhoodie/logging-log4j2/pull/14

**Pull request closed without merging**